admin_manual/configuration_files/encryption_details.rst

```diff
@@ -0,0 +1,356 @@
+==================
+Encryption details
+==================
+
+This document - provided by `SysEleven <https://syseleven.de>`_ - describes the server-side encryption scheme implemented by Nextcloud's default encryption module. This includes:
+
+- the encryption and signature of files with a master key.
+- the encryption and signature of files with a public sharing key.
+- the encryption and signature of files with a recovery key.
+- the encryption and signature of files with a user key.
+
+These conventions apply throughout this document:
+
+- Given file paths in this document are relative to the Nextcloud data directory that can be retrieved as ``datadirectory`` from the ``config.php``.
+- Placeholders are denoted as ``$variable``. The variable has to be replaced with the appropriate information.
+- Static strings are denoted as ``"some string"``.
+- The concatenation of strings is denoted as ``$variable."some string"``.
+
+.. note:: This document describes the server-side encryption scheme as implemented by Nextcloud 16. Previous versions of Nextcloud implemented slightly different schemes which Nextcloud still supports for backwards compatibility. Files that have been encrypted by a recent version of Nextcloud should follow the structure documented below. However, files that have been encrypted by previous versions of Nextcloud may have slightly different structures.
+
+Key type: master key
+--------------------
+
+While the master key encryption had to be enabled explicitly by calling ``./occ encryption:enable-master-key`` in older versions of Nextcloud it is now the default encryption mode in newer versions including Nextcloud 16. With master key encryption enabled there is one central key that is used to secure the files handled by Nextcloud. The master key is protected by a password that can be generated by the server administrator. The advantage of the master key encryption is that the encryption is transparent to the users but has the disadvantage that the server administrator is able to decrypt user files without knowing any user password.
+
+Key type: public sharing key
+----------------------------
+
+The public sharing key is used to secure files that have been publicly shared. The public sharing key is protected by a password that can be generated by the server administrator. The advantage of the public sharing key is that it is independent of the selected encryption mode so that Nextcloud is able to provide publicly shared files to outside parties.
+
+Key type: recovery key
+----------------------
+
+The recovery key is used to provide a restore mechanism in cases where the user key encryption is enabled and users have lost their passwords. The recovery key is protected by a recovery password that the server administrator should store securely. The advantage of the recovery key is that files can be recovered but has the disadvantage that the server administrator is able to decrypt user files without knowing any user password.
```

Comment:
The recovery key is only used if user keys are activated and no master key is used. Both, the admin and the user must enable the recovery key:
https://docs.nextcloud.com/server/stable/admin_manual/configuration_files/encryption_configuration.html#enabling-users-file-recovery-keys